api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -887,6 +898,9 @@ message Cluster {
   // Configuration for load balancing subsetting.
   LbSubsetConfig lb_subset_config = 22;
 
+  // Configuration for the ShuffleShard.
+  LbShuffleShardConfig lb_shuffle_shard_config = 53;
```

Comment:
As I already mentioned previously, we will want to put this into an extension configuration and not add it to the Cluster.